ARCs/arc-0023.md

```diff
@@ -0,0 +1,76 @@
+---
+arc: 23
+title: Sharing Application's information
+description: Append informations to the compiled application's bytes
+author: Stéphane Barroso (@sudoweezy)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/80
+status: Draft
+type: Meta
+created: 2023-01-11
+---
+
+## Abstract
+
+The following document introduces a convention for appending information to the compiled application's bytes. 
+The goal of this convention is to standardize the process of verifying and interacting with smart contracts. The encoded byte is `arc23` followed by the length of the CID representing a folder of files & the CID itself.
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+> Comments like this are non-normative.
+
+### CID
+The <a href="https://github.com/multiformats/cid">CID</a> allows to access the corresponding files using <a href="https://docs.ipfs.tech/">IPFS</a>.
+
+The CID **MUST** represent one file or a folder of files.
+The minimum requirement for this ARC is a file representing the ABI (ABI as described in [ARC-4](arc-0004.md))
```

Comment:
```suggestion
The minimum requirement for this ARC is a folder containing a file `contract.json` representing the contract metadata (as described in the ABI in [ARC-4](arc-0004.md), and as extended by future potential ARCs).
```

Reply:
The above comment is to add the ability to extend the contract metadata.
Also I think ABI is a general term defining methods, ...
while the specific metadata can be called "contract metadata" and named "contract.json".

Reply:
This also follows convention there https://github.com/algorand-devrel/beaker/blob/master/examples/account_storage/DiskHungry.artifacts/contract.json